daemon/mgr/container.go

```diff
@@ -1067,6 +1067,12 @@ func (mgr *ContainerManager) Remove(ctx context.Context, name string, options *t
 		if err := mgr.Client.RemoveSnapshot(ctx, c.ID); err != nil {
 			logrus.Errorf("failed to remove snapshot of container %s: %v", c.ID, err)
 		}
+	} else {
+		// if creating the container by specify a rootfs,
+		// we should umount the mounted rootfs when delete it.
+		if err := mount.Unmount(c.BaseFS, 0); err != nil {
```

Comment:
Please combine the `else` and `if`. I think that is better.

Reply:
Good catch!